UI/src/app/pages/mapping/components/dialog/dialog.component.html

```diff
@@ -0,0 +1,88 @@
+<mat-card class="dialog">
+  <mat-list #shoes>
+    <ng-container *ngFor="let comment of comments">
+      <ng-template 
+      [ngTemplateOutlet]="loadTemplate(comment)" 
+      [ngTemplateOutletContext]="{ $implicit: comment}">
+      </ng-template>
+    </ng-container>
+  </mat-list>
+
+  <input [(ngModel)]="value" [disabled]="isDisabled()" placeholder="Input comment" class="input input-add" (keyup.enter)="add()">
+
+  <button mat-button
+   (click)="add()"
+   [disabled]="isDisabled()"
+   class="button-add"
+    disableRipple="true">
+    Add
+  </button>
+</mat-card>
+
+<ng-template #readOnlyTemplate let-comment>
+  <mat-list-item class="list-item-read">
+    <div class="comment comment-read">
+      <div class="comment-read-head">
+        <span class="comment-read-date">
+          {{comment.date | date:'MMM d'}}
+          <ng-container *ngIf="comment.hasBeenEdited">
+            <span>(edited)</span>
+          </ng-container>
+        </span>
+        <div class="comment-read-head-buttons">
+          <button mat-icon-button disableRipple="true" (click)="edit(comment)">
+            <mat-icon>
+              edit
+            </mat-icon>
+          </button>
+          <button mat-icon-button disableRipple="true" (click)="delete(comment)">
+            <mat-icon>
+              delete
+            </mat-icon>
+          </button>
+        </div>
+      </div>
+      <div class="comment-read-body">
+        <p class="comment-read-text">
+          {{comment.text}}
+        </p>
+      </div>
+    </div>
+  </mat-list-item>
+</ng-template>
+
+<ng-template #editTemplate let-comment>
+  <mat-list-item class="list-item-edit">
```

Comment:
As soon as our styles are incapsulated inside of the component we could use simpler class names  or even use tags if it's possible